libcxx/test/std/ranges/range.adaptors/range.stride.view/ctor.pass.cpp

```diff
@@ -0,0 +1,49 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++03, c++11, c++14, c++17, c++20
+
+// constexpr explicit stride_view(_View, range_difference_t<_View>)
+
+#include <type_traits>
+
+#include "test_convertible.h"
+#include "test_iterators.h"
+#include "types.h"
+
+// There is no default ctor for stride_view.
+using View = BasicTestView<cpp17_input_iterator<int*>>;
+static_assert(!std::is_default_constructible_v<std::ranges::stride_view<View>>);
+
+// Test that the stride_view can only be explicitly constructed.
+static_assert(!test_convertible<std::ranges::stride_view<View>, View, int>());
+
+constexpr bool test() {
+  {
+    int arr[] = {1, 2, 3};
+    // Test that what we will stride over is move only.
+    static_assert(!std::is_copy_constructible_v<MoveOnlyView<cpp17_input_iterator<int*>>>);
+    static_assert(std::is_move_constructible_v<MoveOnlyView<cpp17_input_iterator<int*>>>);
+
+    MoveOnlyView<cpp17_input_iterator<int*>> mov(cpp17_input_iterator<int*>(arr), cpp17_input_iterator<int*>(arr + 3));
```

Comment:
For these constructor tests, we generally try to test the "salient properties" of the object being created after it has been constructed. In this case, the salient properties are the stride, `begin()` and `end()`. So for each construction I would test these 3 things.

I'd also test the different construction patterns in separate block scopes to make it clearer that they are distinct tests.

You could test with 2 different strides.